### DISCUSSION TITLE
Is Drivetrain Efficiency data available on Carla or UE4?

### DISCUSSION THREAD

**Discussion by MurilloGuindani** [carla-simulator/carla#8239] [Q&A] [answered]
Hi I have been testing a longitudinal acceleration model to estimate RSS safe distance in platooning maneuvers using CARLA.

In the literature equations have parameters such as Drivetrain Efficiency and others. I'm wondering where I can find this information for specific blueprints. Reading CARLA code, I was able to find where physics control are set though C++/Pyhton API but got stuck at VehiclePhysicsControl.h. If I want to go deeper on how the engine physics is simulated, what is the next step?

I was not able to find info on vehicle implementation either on CARLA repo or on UE4.  
There are colleagues on my lab researching suspension predictive maintenance that might benefit from this information too.
Thanks in advance!

**Comment by SonglanWang** [comment, marked as answer]
Check physX API which is used by ue4 to simulate the physical engine.

**Reply by MurilloGuindani** [reply]
Thank you for your answer. 

It led me to read [PhysX manual on vehicles](https://docs.nvidia.com/gameworks/content/gameworkslibrary/physx/guide/Manual/Vehicles.html), [PxVehicleEngineData Class Reference](https://docs.nvidia.com/gameworks/content/gameworkslibrary/physx/apireference/files/classPxVehicleEngineData.html), and [Physics vs. PhysX](https://docs.nvidia.com/gameworks/content/gameworkslibrary/physx/guide/Manual/Introduction.html#physics-vs-physx). For who it may concern, the latter states that: 

> "PhysX is best suited for quasi-real time interactive 3D applications where performance and scalability are more important than precision."

In short, specific data on engine performance is absent in the Carla Implementation. Let me know if I'm wrong.

**Comment by SonglanWang** [comment]
Carla is based on UE4. When I checked Carla source code, it adopts the class Awheeledvehicle from UE4 to generate the vehicle. And this class is warpped basing on WheeledVehicleMovementComponent4W(Engine/Plugins/Runtime/PhysXVehicles/Source/PhysXVehicles/Public/WheeledVehicleMovementComponent4W.h). There are a bunch of parameters in this class to define the physical engine. I'm a beginner for carla and ue4. Not sure if this would help you. Hope it make sense.

**Reply by MurilloGuindani** [reply]
Thanks again, it make sense. I'm also a beginner trying to evaluate the simulator for research purposes. 
I forgot to mention that I used [UAssetGUI](https://github.com/atenfyr/UAssetGUI) to check upon the vehicle blueprints and after a careful read haven't found data on driverain efficiency and other parameters my model uses. I assume that the default PhysX values are being used.

**Reply by SonglanWang** [reply]
Sounds good. And epic trying to move from PhysX to Chaos which is a physical engine developed by themselves.

**Reply by MurilloGuindani** [reply]
Thanks for the information. I read the documentation and the transmission ratio includes drivetrain efficiency for simplification.I will check out Chaos when possible. 

Cheers!
